app/src/lib/stores/app-store.ts

```diff
@@ -1621,6 +1625,74 @@ export class AppStore extends TypedBaseStore<IAppState> {
     return true
   }
 
+  private async _triggerConflictsFlow(repository: Repository) {
+    if (enableNewRebaseFlow()) {
+      const repoState = this.repositoryStateCache.get(repository)
+      const { conflictState } = repoState.changesState
+
+      if (conflictState === null) {
+        return
+      }
+
+      if (conflictState.kind === 'merge') {
+        await this.showMergeConflictsDialog(repository, conflictState)
+      } else if (conflictState.kind === 'rebase') {
+        await this.showRebaseConflictsDialog(repository)
+      } else {
+        assertNever(conflictState, `Unsupported conflict kind`)
+      }
+    } else {
+      this._triggerMergeConflictsFlow(repository)
+    }
+  }
+
+  /** display the rebase flow, if not already in this flow */
+  private async showRebaseConflictsDialog(repository: Repository) {
+    // TODO: check the current popup is a rebase conflicts dialog
+    // TODO: if already in flow, exit
+    // TODO: otherwise show the popup
+  }
+
+  /** starts the conflict resolution flow, if appropriate */
+  private async showMergeConflictsDialog(
+    repository: Repository,
+    conflictState: MergeConflictState
+  ) {
+    // are we already in the merge conflicts flow?
+    const alreadyInFlow =
+      this.currentPopup !== null &&
+      (this.currentPopup.type === PopupType.MergeConflicts ||
+        this.currentPopup.type === PopupType.AbortMerge)
+
+    // have we already been shown the merge conflicts flow *and closed it*?
+    const alreadyExitedFlow = this.mergeConflictsBannerState !== null
+
+    if (alreadyInFlow || alreadyExitedFlow) {
+      return
+    }
+
+    const possibleTheirsBranches = await getBranchesPointedAt(
+      repository,
+      'MERGE_HEAD'
+    )
+    // null means we encountered an error
+    if (possibleTheirsBranches === null) {
+      return
+    }
+    const theirBranch =
+      possibleTheirsBranches.length === 1
```

Comment:
This code is duplicating the existing `AppStore._triggerMergeConflictsFlow` function...

https://github.com/desktop/desktop/blob/a23bdc8f6c573b2bffd0a8bbffb53c12cf911edd/app/src/lib/stores/app-store.ts#L1649-L1657

My only change to the flow is to pass in the `conflictState` which is known in the caller function, but because of the feature flag I wanted to preserve the existing flow.

I traced this change to f543e8fba126d89c427e453724856b492ee17021 and #6067, which mentions:

> `git branch --points-at=MERGE_HEAD` works but only if one branch has the same tip, otherwise we get multiple results

So in the case of having zero or multiple branches, we don't show a branch name here to avoid confusing users ("I was on branch A but the app now says branch B, whyyyyy?")